src/windshaft/client.js

```diff
@@ -62,49 +60,72 @@ WindshaftClient.prototype.instantiateMap = function (options) {
     }
   };
 
-  if (this._usePOST(payload, params)) {
-    this._post(payload, params, ajaxOptions);
+  var encodedURL = this._generateEncodedURL(mapDefinition, params);
+  if (!this._isURLTooLong(encodedURL)) {
```

Comment:
I always try to do that, but I think in this case it reads better as it's now given that it's a sequence of options.

```
if (something) {
  // first option
} else {
  if (somethingElse) {
    // second option
  } else {
    // third option
  }
}
```

I thought about different method names to remove the negation in the conditional, but I could only think of names that were not so specific (eg: `_isURLValid`).